src/main/java/cms/gov/madie/measure/services/BundleService.java

```diff
@@ -0,0 +1,60 @@
+package cms.gov.madie.measure.services;
+
+import cms.gov.madie.measure.exceptions.BundleOperationException;
+import cms.gov.madie.measure.exceptions.CqlElmTranslationErrorException;
+import cms.gov.madie.measure.exceptions.CqlElmTranslationServiceException;
+import cms.gov.madie.measure.exceptions.InvalidResourceBundleStateException;
+import com.nimbusds.oauth2.sdk.util.StringUtils;
+import gov.cms.madie.models.measure.ElmJson;
+import gov.cms.madie.models.measure.Measure;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.util.CollectionUtils;
+
+@Slf4j
+@Service
+@AllArgsConstructor
+public class BundleService {
+
+  private final FhirServicesClient fhirServicesClient;
+
+  private final ElmTranslatorClient elmTranslatorClient;
+
+  public String bundleMeasure(Measure measure, String accessToken) {
+    if (measure == null) {
+      return null;
+    }
+
+    if (measure.isCqlErrors()) {
+      throw new InvalidResourceBundleStateException(
+          "Measure", measure.getId(), "since CQL errors exist.");
+    }
+
+    if (CollectionUtils.isEmpty(measure.getGroups())) {
+      throw new InvalidResourceBundleStateException(
+          "Measure", measure.getId(), "since there are no associated measure groups.");
+    }
+
+    if (measure.getElmJson() == null || StringUtils.isBlank(measure.getElmJson())) {
```

Comment:
`isBlank` is null safe and will return `true` when input is null. See [Apache javadocs](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#isBlank-java.lang.CharSequence-)

Reply:
What's the value add of this check when fresh elm is being generated on line 45?

Reply:
Yeah, that is interesting, we don't need it. There is also a check for errors in Elm. I am going to remove it.